torch/csrc/api/src/nn/modules/normalization.cpp

```diff
@@ -0,0 +1,40 @@
+#include <torch/nn/modules/normalization.h>
+
+#include <torch/cuda.h>
+#include <torch/types.h>
+#include <torch/utils.h>
+#include <torch/nn/init.h>
+
+#include <ostream>
+#include <utility>
+
+namespace torch {
+namespace nn {
+
+LayerNormImpl::LayerNormImpl(const LayerNormOptions& options_) : options(options_) {
+  reset();
+}
+
+void LayerNormImpl::reset() {
+  if (options.elementwise_affine()) {
+    weight = register_parameter("weight", torch::ones(torch::IntArrayRef(options.normalized_shape())));
```

Comment:
I think the following should work, because `std::vector<int64_t>` can be converted to `torch::IntArrayRef` implicitly:
```suggestion
    weight = register_parameter("weight", torch::ones(options.normalized_shape()));
```

Reply:
Is there a reason we use `torch::ones`? I believe to match the Python version we should use `torch::empty`.

Reply:
ah I see that you are moving the logic of `reset_parameters` to here. I think to match the Python version better, we can keep the `reset_parameters` logic at the end of the `reset()` call, like how the Python version does it.